### DISCUSSION TITLE
Data volume

### DISCUSSION THREAD

**Discussion by nhatfield** [prometheus/pushgateway#447] [Q&A] [answered]
Which volume do I need to mount for a docker container? I want to be able to recreate my pushgateway without losing all of the data that has already been pushed

**Comment by beorn7** [comment, marked as answer]
You can mount whatever volume you like. But you have to provide the path to a persistence file on that volume using the `--persistence.file=....` flag for the Pushgateway.

**Comment by nhatfield** [comment]
Thank you @beorn7
